### ISSUE TITLE
Grunt task fails

### ISSUE THREAD

**Issue by monolithed** [hapijs/hapi#2358]
Examples from http://hapijs.com/api/8.1.0

```
var Hapi = require('hapi');
var server = new Hapi.Server();
server.connection({ port: 80 });

var handler = function (request, reply) {

    return reply('The page was not found').code(404);
};

server.route({ method: '*', path: '/{p*}', handler: handler });
```

```
Running "hapi:development" (hapi) task
Fatal error: Hapi ["utils/server.js"] -  TypeError: Object #<Object> has no method 'route'
```

```
path   : '/{path*}',
```

```
Running "hapi:development" (hapi) task
Fatal error: Hapi ["utils/server.js"] -  Error: New route /{path*} conflicts with existing /{path*}
```

BTW `node server.js` // Ok

**Comment by hueniverse** [comment]
I don't understand what the problem is.

**Comment by monolithed** [comment]
#### Grunt task

```js
var path = require('path');

module.exports = function (grunt, build) {
	return {
		server: {
			options: {
				server: path.resolve('utils/server.js'),
				noasync: true
			}
		}
	}
};
```

#### Server

```js
var Hapi = require('hapi');

var server = new Hapi.Server();

server.connection({
  host: 'localhost',
  port: 80
});

server.route({
    method: 'GET',
    path: '/{path*}', 
    handler: function (request, reply) {
          return reply('The page was not found').code(404);
    }
});
```

```
➜ grunt hapi

Running "hapi:server" (hapi) task
Fatal error: Hapi ["utils/server.js"] -  Error: New route /{path*} conflicts with existing /{path*}
```

```
➜ node utils/server.js

Server running at: http://localhost:9000
```

**Comment by Marsup** [comment]
Is it normal that you don't export anything in server.js ?

**Comment by hueniverse** [comment]
I don't use grunt. This is better discussed at hapijs/discuss. I'll leave it open for a bit in case someone knows the answer.

**Comment by monolithed** [comment]
@Marsup, of course no, rest of the code:

```js
server.start(function () {
	console.log('Server running at: %s', server.info.uri);
});

module.exports = server;
```

**Comment by AdriVanHoudt** [comment]
Can you post the whole gruntfile + task?

**Comment by monolithed** [comment]
#### Gruntfile

```
var path = require('path'),
	load = require('load-grunt-config');

module.exports = function (grunt) {
	var tasks = path.join(process.cwd(), 'tasks');

	load(grunt, {
		configPath: tasks
	});
};
```


#### Grunt task

```
var path = require('path');

module.exports = function (grunt, build) {
    return {
        server: {
            options: {
                server: path.resolve('utils/server.js'),
                noasync: true
            }
        }
    }
};
```

**Comment by AdriVanHoudt** [comment]
In this (https://github.com/athieriot/grunt-hapi/blob/master/test/fixtures/create_server.js) example they return the server object instead of exporting it?

**Comment by monolithed** [comment]
I do export as written in the documentation
https://github.com/athieriot/grunt-hapi/

```js
// lib/index.js:
var Hapi = require('hapi');
var app = new Hapi.Server();
module.exports = app;
```

**Comment by AdriVanHoudt** [comment]
Yes but the example returns the server object instead of exporting it 

```javascript
var Hapi = require('hapi');
          
var createServer = function(host, port, options) {
  // Create a server with specified options or defaults
  ...
  return server;
  ^^^^^^ 
}

module.exports = function(host, port, options) {
    return createServer(host, port, options);
    ^^^^^^
};
```

**Comment by monolithed** [comment]
```js
var Hapi = require('hapi');

var createServer = function (host, port, options) {
	var server = new Hapi.Server;

	server.connection({
		host: 'localhost',
		port: 9000
	});

	server.route({
		method : 'GET',
		path   : '/{path*}',
		handler: function (request, reply) {
			reply({status: 'ok'});
		}
	});


	return server;
}

module.exports = function (host, port, options) {
	return createServer(host, port, options);
};
```

```
➜ grunt hapi
Running "hapi:development" (hapi) task
Fatal error: Hapi ["/Users/utils/server.js"] -  Error: New route /{path*} conflicts with existing /{path*}
```

Have you tried to run your example with the Grunt?

**Comment by AdriVanHoudt** [comment]
@monolithed I have not tried it myself, that example came from the grunt-hapi docs. Maybe open up an issue there?

**Comment by monolithed** [comment]
@AdriVanHoudt, perhaps you're right.  https://github.com/athieriot/grunt-hapi/issues/11

**Comment by athieriot** [comment]
Hi there, 

   I am the maintener of grunt-hapi.

   The error occurs because the plugin is creating a default route on ```/{path*}``` configure to expose static files.
   It therefore conflicts with your own root path handler.
   This has already been reported here: https://github.com/athieriot/grunt-hapi/issues/9

   The quick fix for you will be to override the option ```bases``` with ```{ }```
   In your case:

   ```javascript
     server: {
         options: {
                bases: {},
                server: path.resolve('utils/server.js'),
                noasync: true
         }
     }
   ```

   More generaly now, for some reasons, it used to be a good idea at the time.
   Obviously this is causing trouble now, so I am gonna release a version with a more sensitive default.

Hope this is all right and if there's anything __grunt-hapi__ should do differently, don't hesite to feel an issue.

Aurélien

**Comment by monolithed** [comment]
Thanks, the problem was solved in 0.9.1 :v:
